spec/consensus/bft-time.md

```diff
@@ -3,54 +3,120 @@ order: 2
-corresponds to the number of milliseconds since January 1, 1970.
-Before defining rules that need to be enforced by Tendermint, the consensus algorithm adopted in CometBFT,
-so the properties above holds, we introduce the following definition:
-
-- median of a Commit is equal to the median of `Vote.Time` fields of the `Vote` messages,
-where the value of `Vote.Time` is counted number of times proportional to the process voting power. As
-the voting power is not uniform (one process one vote), a vote message is actually an aggregator of the same votes whose
-number is equal to the voting power of the process that has casted the corresponding votes message.
-
-Let's consider the following example:
-
-- we have four processes p1, p2, p3 and p4, with the following voting power distribution (p1, 23), (p2, 27), (p3, 10)
-and (p4, 10). The total voting power is 70 (`N = 3f+1`, where `N` is the total voting power, and `f` is the maximum voting
-power of the faulty processes), so we assume that the faulty processes have at most 23 of voting power.
-Furthermore, we have the following vote messages in some LastCommit field (we ignore all fields except Time field):
-    - (p1, 100), (p2, 98), (p3, 1000), (p4, 500). We assume that p3 and p4 are faulty processes. Let's assume that the
-      `block.LastCommit` message contains votes of processes p2, p3 and p4. Median is then chosen the following way:
-      the value 98 is counted 27 times, the value 1000 is counted 10 times and the value 500 is counted also 10 times.
-      So the median value will be the value 98. No matter what set of messages with at least `2f+1` voting power we
-      choose, the median value will always be between the values sent by correct processes.
-
-We ensure Time Monotonicity and Time Validity properties by the following rules:
-  
-- let rs denotes `RoundState` (consensus internal state) of some process. Then
-`rs.ProposalBlock.Header.Time == median(rs.LastCommit) &&
-rs.Proposal.Timestamp == rs.ProposalBlock.Header.Time`.
-
-- Furthermore, when creating the `vote` message, the following rules for determining `vote.Time` field should hold:
-
-    - if `rs.LockedBlock` is defined then
-    `vote.Time = max(rs.LockedBlock.Timestamp + time.Millisecond, time.Now())`, where `time.Now()`
-        denotes local Unix time in milliseconds
-
-    - else if `rs.Proposal` is defined then
-    `vote.Time = max(rs.Proposal.Timestamp + time.Millisecond,, time.Now())`,
-
-    - otherwise, `vote.Time = time.Now())`. In this case vote is for `nil` so it is not taken into account for
-    the timestamp of the next block.
+BFT Time is a Byzantine fault-tolerant algorithm for computing [block times](./time.md).
+
+> :warning:
+> CometBFT `v1.x` introduced [Proposer-Based Timestamps (PBTS)][pbts-spec],
+> intended to be a replacement for BFT Time.
+> Users are strongly encouraged to adopt PBTS in new chains, or when upgrading
+> existing chains.
```

Comment:
I'd add something like:

_BFT-Time MAY be deprecated in a future version of CometBFT_

... even if we know it's not likely to happen, as it would be a mess for blocksync